src/pairs/DPSolver.py

```diff
@@ -0,0 +1,20 @@
+from src.pairs.util import intMapper
+import numpy as np
+class DPSolver:
+    def __init__(self, game):
+        deck_states = intMapper(game.deck.values(), game.deck.values())
+        player_states = intMapper(game.deck.values(), [1 for _ in range(len(game.deck.values()))])
+        dim = [player_states] * game.numPlayers
+        [deck_states].extend(dim)
+        dim.extend([game.numPlayers])
+        self.dp_table = np.zeros((tuple(dim)))
```

Comment:
You can remove a pair of redundant parentheses.

Reply:
done